celery_batches/__init__.py

```diff
@@ -285,15 +287,36 @@ def apply(
         return super().apply(([request],), {}, *_args, **options)
 
     def _do_flush(self) -> None:
-        logger.debug("Batches: Wake-up to flush buffer...")
-        requests = None
-        if self._buffer.qsize():
-            requests = list(consume_queue(self._buffer))
-            if requests:
-                logger.debug("Batches: Buffer complete: %s", len(requests))
-                self.flush(requests)
-        if not requests:
-            logger.debug("Batches: Canceling timer: Nothing in buffer.")
+        logger.debug("Batches: Wake-up to flush buffers...")
+
+        ready_requests = []
+        pending_requests = []
+
+        all_requests = list(consume_queue(self._buffer)) + list(
+            consume_queue(self._pending)
+        )
+        for req in all_requests:
+            if req.eta is not None:
+                if req.eta <= datetime.now(timezone.utc):
+                    # ETA has elapsed, request is ready
+                    ready_requests.append(req)
+                else:
+                    # ETA has not elapsed, add to pending queue
+                    pending_requests.append(req)
```

Comment:
I don't think `pending_requests` is needed, we've already emptied `self._pending` on line 296, so I think this can be:

```suggestion
                    self._pending.put(req)
```

Which means we don't need to re-iterate over `pending_requests` later on.